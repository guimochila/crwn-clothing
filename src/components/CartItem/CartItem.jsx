import React from 'react';
import {
  CartItemContainer,
  CartItemImage,
  ItemDetailsContainer,
  ItemName,
} from './CartItem.styles';

function CartItem({ item: { imageUrl, price, name, quantity } }) {
  return (
    <CartItemContainer>
      <CartItemImage src={imageUrl} alt={name} />
      <ItemDetailsContainer>
        <ItemName>{name}</ItemName>
        <span>
          {quantity} x ${price}
        </span>
      </ItemDetailsContainer>
    </CartItemContainer>
  );
}

export default React.memo(CartItem);
