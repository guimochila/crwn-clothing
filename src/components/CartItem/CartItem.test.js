import React from 'react';
import { shallow } from 'enzyme';

import CartItem from './CartItem';

describe('CartItem Component', () => {
  test('it should render CartItem', () => {
    const mockProps = {
      item: {
        imageUrl: 'http://imgsrc.com/image',
        price: 21,
        name: 'Testing',
        quantity: 1,
      },
    };

    const wrapper = shallow(<CartItem {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
