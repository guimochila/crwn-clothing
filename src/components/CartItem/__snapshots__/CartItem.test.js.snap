// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component it should render CartItem 1`] = `
<styled.div>
  <styled.img
    alt="Testing"
    src="http://imgsrc.com/image"
  />
  <styled.div>
    <styled.span>
      Testing
    </styled.span>
    <span>
      1
       x $
      21
    </span>
  </styled.div>
</styled.div>
`;
